### PR TITLE
Add VS Code devcontainer config

### LINKED ISSUE #173: VSCode Devcontainer configuration
This would be nice since we use it on quite a lot of projects – should be basic, but worthwhile nonetheless.

### PR DESCRIPTION
**Pull request by bcdickinson**
## Description

Adds a simple VS Code devcontainer setup for this project based on the existing docker-compose setup.

Closes #173 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] New and existing unit tests pass locally with my changes
- [x] I have added an appropriate CHANGELOG entry

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -2,48 +2,68 @@
...
-poetry install
-# Start the server for testing:
-poetry run django-admin runserver --settings=tests.settings.dev --pythonpath=.
-# Or to try out the render_patterns command:
-poetry run django-admin render_patterns --settings=tests.settings.dev --pythonpath=. --dry-run --verbosity 2
-```
+For users of Docker and VS Code, there is a [devcontainer](https://code.visualstudio.com/docs/remote/containers) setup included in the repository
+that will automatically install the Python dependencies and start the frontend tooling.
+
+Once the container is built, open a terminal with VS Code and run `django-admin runserver` and click the URL (normally http://127.0.0.1:8000/) to open the app in your browser. You'll see a 404 page, because there's nothing at `/`, add `pattern-libary/` to the end of the URL to view the demo app.
```
Comment (thibaudcolas) [inline CONTRIBUTING.md:29]: Should we change our dev site to have the pattern library at the root?

File: Dockerfile [hunk 2]
```diff
@@ -1,9 +1,9 @@
-FROM python:3.7
+FROM python:3.10
 
 WORKDIR /app
 
-RUN useradd dpl && \
-    pip install --pre poetry && \
+RUN useradd --create-home dpl && \
+    pip install "poetry>=1.1.2,<2" && \
```
Comment (thibaudcolas) [inline Dockerfile:6]: I had trouble with Python 3.10 + Poetry versions older than 1.1.12. Could be worth specifying it as the minimum here?

**Review by thibaudcolas — APPROVED** [review]
@bcdickinson gave this a quick look, all seems fine to me. I’ll try running this locally once it’s merged 😬 .

File: Dockerfile (hunk 2 above)
Comment (bcdickinson) [reply Dockerfile:6]: I meant to do this, actually, this is a typo :facepalm:

File: CONTRIBUTING.md (hunk 1 above)
Comment (bcdickinson) [reply CONTRIBUTING.md:29]: Yes, good idea.

**Pull request merged**